metafacture-elasticsearch/src/main/java/org/metafacture/elasticsearch/JsonToElasticsearchBulk.java

```diff
@@ -116,7 +126,7 @@ public void process(String obj) {
             Map<String, Object> detailsMap = new HashMap<String, Object>();
             Map<String, Object> indexMap = new HashMap<String, Object>();
             indexMap.put("index", detailsMap);
-            detailsMap.put("_id", findId(json));
+            if (idPath.length > 0) detailsMap.put("_id", findId(json));
```

Comment:
May I ask why you want to exclude the `_id` field from the bulk metadata? A null value (`"_id":null`) will trigger the automatic ID generation.

Reply:
Ah, didn't know that!
 But there is no way to instantiate an object by passing only two parameters, I think. Thus from user perspective it's hard to know about that (as like me) and somewhat counter intuitive to pass a `null` as `keyId` to get one generated by Elasticsearch.
Also it blows the produced `bulk.ndjson` (a bit).

Reply:
I don't object to adding the new constructor. I was just wondering why you made the actual implementation changes. The obvious advantage of not changing the implementation would be that the tests wouldn't have to be fixed ;)

Reply:
Ok I see :) and yes I am struggling right now to make these tests pass ...

Reply:
If you're intent on dropping the `_id` field from the output, I can take a look at the tests. Would you mind me pushing into your branch?

Reply:
I wouldn't mind at all, rather thank you :)

Reply:
See PR #324.